app/src/main/java/org/sagebionetworks/research/mpower/studyburst/StudyBurstViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.sagebionetworks.research.mpower.studyburst
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import org.sagebionetworks.research.mpower.R
+
+class StudyBurstViewModel : ViewModel() {
+
+    private val title = MutableLiveData<String>()
+    private val message = MutableLiveData<String>()
+    private val expires = MutableLiveData<String>()
+    private var items = MutableLiveData<List<StudyBurstItem>>()
+
+    fun init() {
+        title.setValue("Great start!")
+        message.setValue("Complete the rest of todays activities within the next hour in order to complete the study burst for Day 12.")
```

Comment:
The swift code for this String function is...
```

        let messageStr: String? = {
            if studyBurstManager.missedDaysCount == 0 {
                // The message will vary by the current day of the study burst
                let format = String(format: "STUDY_BURST_MESSAGE_DAY_%@", currentDaysStr)
                return Localization.localizedString(format)
            }
            else {
                // The message will be the same for each day of the study burst and will simply
                // indicate the current day and the number of missed days
                let missedDaysStr = formatter.string(for: studyBurstManager.missedDaysCount)!
                
                let format = studyBurstManager.missedDaysCount > 1 ?
                    Localization.localizedString("STUDY_BURST_MESSAGE_IN_%@_DAYS_MISSED_%@_DAYS") :
                    Localization.localizedString("STUDY_BURST_MESSAGE_IN_%@_DAYS_MISSED_ONE_DAY")
                
                let str = studyBurstManager.missedDaysCount > 1 ?
                    String.localizedStringWithFormat(format, currentDaysStr, missedDaysStr) :
                    String.localizedStringWithFormat(format, currentDaysStr)
                
                return str
            }
        }()

```

and also the localized strings used are:

```
"STUDY_BURST_MESSAGE_DAY_%@" = Activities should be done every day and ideally all at once. You can begin them at any point during the day. ";
"STUDY_BURST_MESSAGE_IN_%@_DAYS_MISSED_%@_DAYS" = "In %1$@ days you’ve only missed %2$@ Study Burst days!";
"STUDY_BURST_MESSAGE_IN_%@_DAYS_MISSED_ONE_DAY" = "In %1$@ days you’ve only missed 1 Study Burst day!";

```

Reply:
👍